dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -346,73 +346,64 @@ protected void githubWebhookDelete(String repository, String gitReference, Strin
-            lambdaEventDAO.create(lambdaEvent);
-            sessionFactory.getCurrentSession().getTransaction().commit();
+            messageWriter.println(msg);
+            messageWriter.println("Terminated processing of .dockstore.yml.");
+
             throw new CustomWebApplicationException(msg, statusCodeForLambda(ex));
-        } catch (Exception ex) {
-            endRateLimit = gitHubSourceCodeRepo.getGhRateLimitQuietly();
+
+        } finally {
+
+            // Make an entry in the github apps logs.
+            final boolean finalIsSuccessful = isSuccessful;
+            new TransactionHelper(sessionFactory).transaction(() -> {
+                LambdaEvent lambdaEvent = createBasicEvent(repository, gitReference, username, LambdaEvent.LambdaEventType.PUSH);
+                lambdaEvent.setSuccess(finalIsSuccessful);
+                setEventMessage(lambdaEvent, stringWriter.toString());
+                lambdaEventDAO.create(lambdaEvent);
+            });
+
+            GHRateLimit endRateLimit = gitHubSourceCodeRepo.getGhRateLimitQuietly();
             gitHubSourceCodeRepo.reportOnGitHubRelease(startRateLimit, endRateLimit, repository, username, gitReference, isSuccessful);
-            String msg = "User " + username + ": Unhandled error while handling push event for repository " + repository + " and reference " + gitReference + "\n" + ex.getMessage();
-            LOG.error(msg, ex);
-            rollbackAndBeginNewTransaction();
-            LambdaEvent lambdaEvent = createBasicEvent(repository, gitReference, username, LambdaEvent.LambdaEventType.PUSH);
-            lambdaEvent.setSuccess(false);
-            lambdaEvent.setMessage(ex.getMessage());
-            lambdaEventDAO.create(lambdaEvent);
-            sessionFactory.getCurrentSession().getTransaction().commit();
-            throw new CustomWebApplicationException(msg, LAMBDA_FAILURE);
+        }
+
+        if (!isSuccessful) {
+            throw new CustomWebApplicationException("At least one entry in .dockstore.yml could not be registered.", LAMBDA_FAILURE);
         }
     }
 
-    /**
-     * Rollback current DB transaction then start a new one so we can record a lambda event.
-     * https://github.com/dockstore/dockstore/issues/4434
-     */
-    private void rollbackAndBeginNewTransaction() {
-        final Transaction transaction = sessionFactory.getCurrentSession().getTransaction();
-        final boolean isActive = transaction.isActive();
-        final boolean canRollback = transaction.getStatus().canRollback();
-        if (isActive && canRollback) {
-            transaction.rollback();
-        } else {
-            // I don't think this should ever happen
-            LOG.error("Transaction is not active or cannot be rolled back. Active: {}; Can rollback: {}", isActive, canRollback);
+    private void setEventMessage(LambdaEvent lambdaEvent, String message) {
+        message = StringUtils.stripEnd(message, "\n");
```

Comment:
We discussed this in another PR and I know you disagree, but on reassigning parameters:

https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#avoidreassigningparameters

https://checkstyle.sourceforge.io/config_coding.html#ParameterAssignment